rllib/env/single_agent_episode.py

```diff
@@ -0,0 +1,525 @@
+import numpy as np
+import uuid
+
+from gymnasium.core import ActType, ObsType
+from typing import Any, Dict, List, Optional, SupportsFloat
+
+from ray.rllib.policy.sample_batch import SampleBatch
+
+
+class SingleAgentEpisode:
+    def __init__(
+        self,
+        id_: Optional[str] = None,
+        *,
+        observations: List[ObsType] = None,
+        actions: List[ActType] = None,
+        rewards: List[SupportsFloat] = None,
+        infos: List[Dict] = None,
+        states=None,
+        t_started: int = None,
+        is_terminated: bool = False,
+        is_truncated: bool = False,
+        render_images: List[np.ndarray] = None,
```

Comment:
nit: `Optional[List[np.ndarray]]`

Reply:
ANother great one :) Resolved in next commit.